### PR TITLE
Transform sequences

### PR DESCRIPTION
**Pull request by peppy**

### CODE REVIEW

File: osu.Framework/Graphics/Drawable.cs [hunk 1]
```diff
@@ -1801,6 +1801,26 @@ public virtual Drawable DelayReset()
...
+        /// <summary>
+        /// /// Start a sequence of transforms from an absolute time value.
+        /// </summary>
+        /// <param name="delay">The offset in milliseconds from current time. Note that this stacks with other nested sequences.</param>
+        /// <param name="recursive">Whether this should be applied to all children.</param>
+        /// <returns>A <see cref="TransformSequence"> to be used in a using() statement.</returns>
+        public TransformSequence BeginAbsoluteSequence(double startOffset = 0, bool recursive = false)
+        {
+            if (transformDelay != 0) throw new InvalidOperationException($"Cannot use {nameof(BeginAbsoluteSequence)} with a non-zero transform delay already present");
+            return new TransformSequence(this,(Clock?.CurrentTime ?? 0) + startOffset, recursive);
```
Comment (thomastanck) [inline osu.Framework/Graphics/Drawable.cs]: Shouldnt it be startOffset - (Clock?.CurrentTime ?? 0) ?

Comment (peppy) [inline osu.Framework/Graphics/Drawable.cs]: yep, well spotted

File: osu.Framework/Graphics/Drawable.cs [hunk 2]
```diff
@@ -2182,4 +2202,47 @@ public enum FillMode
         /// </summary>
         Stretch
     }
+
+    /// <summary>
+    /// A disposable-pattern object to handle isolated sequences of transforms. Should only be used in using blocks.
+    /// </summary>
+    public class TransformSequence : IDisposable
```
Comment (smoogipooo) [inline osu.Framework/Graphics/Drawable.cs]: Move to separate file.

Comment (peppy) [inline osu.Framework/Graphics/Drawable.cs]: even though it's only ever instantiated by Drawable?

doing it this way means we can make the delay properties eventually private.

Comment (smoogipooo) [inline osu.Framework/Graphics/Drawable.cs]: Hmm okay

File: osu.Framework/Graphics/Drawable.cs [hunk 3]
```diff
@@ -1801,6 +1801,26 @@ public virtual Drawable DelayReset()
...
+        /// <summary>
+        /// Start a sequence of transforms with a (cumulative) relative delay applied.
+        /// </summary>
+        /// <param name="delay">The offset in milliseconds from current time. Note that this stacks with other nested sequences.</param>
+        /// <param name="recursive">Whether this should be applied to all children.</param>
+        /// <returns>A <see cref="TransformSequence" /> to be used in a using() statement.</returns>
+        public TransformSequence BeginDelayedSequence(double delay, bool recursive = false) => new TransformSequence(this, delay, recursive);
+
+        /// <summary>
+        /// /// Start a sequence of transforms from an absolute time value.
```
Comment (thomastanck) [inline osu.Framework/Graphics/Drawable.cs]: Perhaps add that these can only be used at the top level (no recursive delays from parent, cannot be inside another BeginDelayedSequence or BeginAbsoluteSequence), or it'll throw?

**Comment by peppy** [comment]
don't merge juuust yet, checking codefactor issue

**Pull request merged**